dice_ml/explainer_interfaces/explainer_base.py

```diff
@@ -87,6 +87,20 @@ def generate_counterfactuals(self, query_instances, total_CFs, desired_class="op
             cf_examples_arr.append(res)
         return CounterfactualExplanations(cf_examples_list=cf_examples_arr)
 
+    def check_query_instance_validity(self, features_to_vary, query_instance):
+        for feature in self.data_interface.feature_names:
+            if feature not in features_to_vary:
+                if feature in self.data_interface.continuous_feature_names:
+                    if not self.feature_range[feature][0] <= query_instance[feature].values[0] <= self.feature_range[feature][1]:
```

Comment:
need to check something more. that permitted_range is actually given by user or not. otherwise feature_range is simply the max min value in the training set which can be violated by query